ansible/roles/test/tasks/pfc_wd/functional_test/functional_test_storm.yml

```diff
@@ -0,0 +1,158 @@
+   pfc_wd_test_port_id: "{{item.value.test_port_id}}"
+   pfc_wd_rx_port_id: "{{item.value.rx_port_id}}"
+   port_type: "{{item.value.test_port_type}}"
+
+- name: Set fact for test port ids and include portchannel other members if exists
+  set_fact:
+    pfc_wd_test_port_ids: "[{{item.value.test_portchannel_members | join(' ')}}]"
+  when: port_type == "portchannel"
+
+- set_fact:
+    pfc_wd_test_port_ids: "[{{pfc_wd_test_port_id}}]"
+  when: port_type in ["vlan", "interface"]
+
+- name: Remove existing IPs from PTF host
+  script: roles/test/files/helpers/remove_ip.sh
+  delegate_to: "{{ptf_host}}"
+  when: port_type == "vlan"
+
+- name: "Set {{pfc_wd_test_neighbor_addr}} to eth{{pfc_wd_test_port_id}} on PTF host"
+  shell: ifconfig eth{{pfc_wd_test_port_id}} {{pfc_wd_test_neighbor_addr}}
+  delegate_to: "{{ptf_host}}"
+  when: port_type == "vlan"
+
+- name: "Update ARP entry on DUT"
+  shell: ping {{minigraph_vlan_interfaces[0]['addr']}} -c 10
+  delegate_to: "{{ptf_host}}"
+  when: port_type == "vlan"
+
+- name: "Update ARP entry on DUT"
+  shell: docker exec -i swss arping {{pfc_wd_test_neighbor_addr}} -c 5
+  when: port_type == "vlan"
+
+- conn_graph_facts: host={{ peer_device }}
+  connection: local
+  become: no
+
+- name: Prepare variables required for PFC test
+  set_fact:
+    pfc_queue_index: 4
+    pfc_frames_number: 100000000
+    pfc_wd_test_pkt_count: 100
+    pfc_fanout_interface: "{{neighbors[pfc_wd_test_port]['peerport']}}"
+    peer_hwsku: "{{device_info['HwSku']}}"
+    peer_mgmt: "{{device_info['mgmtip']}}"
+    testname: functional_test
+
+- set_fact:
+    class_enable: "{{(1).__lshift__(pfc_queue_index)}}"
+
+- set_fact:
+      peer_login: "{{switch_login[hwsku_map[peer_hwsku]]}}"
+
+- name: set pfc storm templates based on fanout platform sku
+  include: roles/test/tasks/pfc_wd/functional_test/set_pfc_storm_templates.yml
+
+
+- block:
+    - set_fact:
+          pfc_gen_file: pfc_gen.py
+
+    - name: Deploy pfc packet generater file to fanout switch
+      include: roles/test/tasks/pfc_wd/functional_test/deploy_pfc_pktgen.yml
+
+    - name: copy the test to ptf container
+      copy: src=roles/test/files/ptftests dest=/root
+      delegate_to: "{{ptf_host}}"
+
+    # 1. Verify storm is detected via syslog entry analysis on the dut
+    - set_fact:
+        test_expect_file: "expect_pfc_wd_detect"
+        test_ignore_file: "ignore_pfc_wd_messages"
+
+    - name: Initialize loganalyzer
+      include: roles/test/files/tools/loganalyzer/loganalyzer_init.yml
+
+    - name: Generate PFC storm on fanout switch
+      action: apswitch template="{{pfc_wd_storm_template}}"
+      args:
+        host: "{{peer_mgmt}}"
+        login: "{{peer_login}}"
+      connection: switch
+      register: pid
+
+    - debug: msg="{{pid}}"
+
+    - name: Let PFC storm happen for a while
+      pause:
+        seconds: 5
+
+    - name: Check if logs contain message that PFC WD detected storm
```

Comment:
You are absolutely correct.

Reply:
Updated the command to be `ansible-playbook test_sonic.yml -i inventory --limit ${SWITCH} --vault-password-file=./vault -e testbed_name=${SWITCH}-${TOPO} -e testcase_name=pfc_wd -e warm_reboot_test=true`

Reply:
> You are absolutely correct in the command. We assume the `pfcwd` has the `start_default` setting configured on the dut.

Is 'start_default' configured by minigraph generation or it is set when public image is built? Otherwise we cannot use the test against public image without additional change to start pfcwd